matter_server/server/ota/__init__.py

```diff
@@ -0,0 +1,55 @@
+"""OTA implementation for the Matter Server."""
+
+from matter_server.server.ota import dcl
+
+HARDCODED_UPDATES: dict[tuple[int, int], dict] = {
+    # OTA requestor example app, useful for testing
+    (0xFFF1, 0x8001): {
+        "vid": 0xFFF1,
+        "pid": 0x8001,
+        "softwareVersion": 2,
+        "softwareVersionString": "2.0",
+        "cdVersionNumber": 1,
+        "softwareVersionValid": True,
+        "otaChecksum": "7qcyvg2kPmKZaDLIk8C7Vyteqf4DI73x0tFZkmPALCo=",
+        "otaChecksumType": 1,
+        "minApplicableSoftwareVersion": 1,
+        "maxApplicableSoftwareVersion": 1,
+        "otaUrl": "https://github.com/agners/matter-linux-example-apps/releases/download/v1.3.0.0/chip-ota-requestor-app-x86-64.ota",
+        "releaseNotesUrl": "https://github.com/agners/matter-linux-example-apps/releases/tag/v1.3.0.0",
+    },
+    (0x143D, 0x1001): {
+        "vid": 0x143D,
+        "pid": 0x1001,
+        "softwareVersion": 10010011,
+        "softwareVersionString": "1.1.11-c85ba1e-dirty",
+        "cdVersionNumber": 1,
+        "softwareVersionValid": True,
+        "otaChecksum": "x2sK9xjVuGff0eefYa4cporDO+Z+WVxxw+JP5Ol+5og=",
+        "otaChecksumType": 1,
+        "minApplicableSoftwareVersion": 10010000,
+        "maxApplicableSoftwareVersion": 10010011,
+        "otaUrl": "https://raw.githubusercontent.com/ChampOnBon/Onvis/master/S4/debug.ota",
+    },
+}
```

Comment:
Question 1: Do you think its a good idea to have this update here as it has a big fat "experimental" vibe around it ;-)

Question 2: So we could extend this list with other hardcoded paths ?

Reply:
> Question 1: Do you think its a good idea to have this update here as it has a big fat "experimental" vibe around it ;-)

It is what their official app distributes too :man_shrugging:  Just much easier to install compared to their update procedure :see_no_evil: 

FWIW, the firmware seems more reliable, at least my S4 switch stayed online since I've updated.

> 
> Question 2: So we could extend this list with other hardcoded paths ?

Yes we can. But it is hard coded, so I don't think this will scale well. I'd definitely prefer if folks would just use the DCL :see_no_evil:

Reply:
Yes, DCL is currently being discussed - te procedures will be simplified soon(ish) but until then we will have to get this going with (temporary) hardcoded lists.

Reply:
I've removed the hardcoded updates now and replaced it with capability to load json files from the update directory. This is currently really meant for development. It shouldn't interfere for user updates, which at this point still just supports the main net DCL.